compiler/resolution/functionResolution.cpp

```diff
@@ -463,6 +463,35 @@ resolveUninsertedCall(Type* type, CallExpr* call) {
   return call->isResolved();
 }
 
+static void resolveFnForCall(FnSymbol* fn, CallExpr* call)
```

Comment:
I'm the opposite here.  While I like reading code this way, and generally write and review Chapel code this way, with C code, I prefer a bottom-up code ordering because I hate having to maintain two copies of a function prototype for something that's local to a file.  For files that I completely own and can maintain (and I realize function resolution isn't one of them), I try to create "tiers" of routines that correspond roughly to how deep they are in the callchain with the entry point at the bottom of the file.  Unfortunately, our compiler is pretty consistently inconsistent on this point, so it's hard to argue that either is dominant (and, in the past, certain team members have gone off and rearranged files to their liking and others' displeasure without much buy-in).  I guess point being, I've learned to live with either style (but still hate it when I change a static routine and have to change two places...)

I'm wondering if simply adding a comment to the routine to answer Mike's "what is it for" question would be a sufficient compromise.